tools/rostopic/src/rostopic/__init__.py

```diff
@@ -207,7 +207,107 @@ def _rostopic_hz(topic, window_size=-1, filter_expr=None):
     while not rospy.is_shutdown():
         _sleep(1.0)
         rt.print_hz()
-    
+
+
+class ROSTopicDelay(object):
+
+    def __init__(self, window_size):
+        import threading
+        self.lock = threading.Lock()
+        self.last_msg_tn = 0
+        self.msg_t0 = -1.
+        self.msg_tn = 0
+        self.delays = []
+
+        # can't have infinite window size due to memory restrictions
+        if window_size < 0:
+            window_size = 50000
+        self.window_size = window_size
+
+    def callback_delay(self, msg):
+        if msg._has_header is False:
```

Comment:
Why the explicit comparison against `False`? Wouldn't `if not msg._has_header` not work?